### DISCUSSION TITLE
Dynamic adding attribute and markup to custom shape

### DISCUSSION THREAD

**Discussion by darekdajcz** [clientIO/joint#2211] [Q&A]
### Introduction

I am using a project with bots in Angular. I need functionality for my shape that will dynamically add attributes and markup to the active shape on the graph. Is there such a possibility if there was no markup for this view in the shape before? It would be best if the attribute could capture an array of this argument and print it column-wise in the shape when adding."

### Steps to reproduce

example below
![image](https://github.com/clientIO/joint/assets/48740052/95a4afc8-cacd-4a85-a8fb-d735a75e80ff)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You can use the `Record` shape from JointJS+ - https://resources.jointjs.com/docs/rappid/v3.7/shapes.standard.html#Record

Record examples:
https://www.jointjs.com/demos/database
https://www.jointjs.com/demos/uml-class-shape-inspector
https://www.jointjs.com/demos/data-mapping

Other examples with list elements:
https://www.jointjs.com/demos/dialog-generator
https://www.jointjs.com/demos/list
https://www.jointjs.com/demos/counters
